src/CoreWCF.WebHttp/src/CoreWCF/Description/WebHttpBehavior.cs

```diff
@@ -0,0 +1,1356 @@
+            return messageDescription;
+        }
+
+        internal static bool SupportsJsonFormat(OperationDescription od)
+        {
+            // if the type is XmlSerializable, then we cannot create a json serializer for it
+            DataContractSerializerOperationBehavior dcsob = ((KeyedByTypeCollection<IOperationBehavior>)od.OperationBehaviors).Find<DataContractSerializerOperationBehavior>();
+            return (dcsob != null);
+        }
+
+        internal static void ValidateIsWebHttpBinding(ServiceEndpoint serviceEndpoint, string behaviorName)
+        {
+            Binding binding = serviceEndpoint.Binding;
+            if (binding.Scheme != "http" && binding.Scheme != "https")
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(
+                    SR.Format(SR.WCFBindingCannotBeUsedWithUriOperationSelectorBehaviorBadScheme,
+                    serviceEndpoint.Contract.Name, behaviorName)));
+            }
+
+            if (binding.MessageVersion != MessageVersion.None)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(
+                    SR.Format(SR.WCFBindingCannotBeUsedWithUriOperationSelectorBehaviorBadMessageVersion,
+                    serviceEndpoint.Address.Uri.AbsoluteUri, behaviorName)));
+            }
+
+            TransportBindingElement transportBindingElement = binding.CreateBindingElements().Find<TransportBindingElement>();
+            if (transportBindingElement != null && !transportBindingElement.ManualAddressing)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(
+                    SR.Format(SR.ManualAddressingCannotBeFalseWithTransportBindingElement,
+                    serviceEndpoint.Address.Uri.AbsoluteUri, behaviorName, transportBindingElement.GetType().Name)));
+            }
+        }
+
+        internal WebMessageBodyStyle GetBodyStyle(OperationDescription od)
+        {
+            WebGetAttribute wga = ((KeyedByTypeCollection<IOperationBehavior>)od.OperationBehaviors).Find<WebGetAttribute>();
+            WebInvokeAttribute wia = ((KeyedByTypeCollection<IOperationBehavior>)od.OperationBehaviors).Find<WebInvokeAttribute>();
+            EnsureOk(wga, wia, od);
+            if (wga != null)
+            {
+                return wga.GetBodyStyleOrDefault(DefaultBodyStyle);
+            }
+            else if (wia != null)
+            {
+                return wia.GetBodyStyleOrDefault(DefaultBodyStyle);
+            }
+            else
+            {
+                return DefaultBodyStyle;
+            }
+        }
+
+        // TODO: Can I remove this?
+        //internal IClientMessageFormatter GetDefaultClientFormatter(OperationDescription od, bool useJson, bool isWrapped)
```

Comment:
Yes, can be removed.

Reply:
This was addressed in b54a2808ec6f4245ccf662c73a2b441ea6c85814.